app/helpers/authentication/active_record_helpers.rb

```diff
@@ -6,32 +6,41 @@ def self.included(base)
 
     module ClassMethods
       def find_or_create_for_github_oauth(auth)
-        find_for_github_oauth(auth) || create_for_github_oauth(auth)
+        find_and_update_for_github_oauth(auth) || create_for_github_oauth(auth)
+      end
+
+      def find_and_update_for_github_oauth(auth)
+        user = find_for_github_oauth(auth)
+        update_user(user, auth) if user && user.email.nil? && user.unconfirmed_email.nil?
```

Comment:
I used `unconfirmed_user?` for now. since I'm not 100% sure what this check is for. 
@bitboxer could you help me out with the naming here?